implemented fibonacci function

vinilapeddireddy opened the pull request:
Fixes #

#### Changes proposed in this PR:

-
-

Commit:
implemented gibonacci function

Commit:
corrected function name

anzdaddy commented on 01_fibonacci/fibi.go:
```diff
@@ -0,0 +1,29 @@
+	}
+	if n >= 1 {
+		fmt.Fprintln(out, b)
+	}
```
You can eliminate the above tests and `fmt.Fprintln` calls if you carefully choose the initial loop conditions and change what you print inside the loop.

vinilapeddireddy replied:
Modified

anzdaddy commented on 01_fibonacci/fibi.go:
```diff
@@ -0,0 +1,29 @@
+	for i := 2; i <= n; i++ {
+		fmt.Fprintln(out, a+b)
+		b = a + b
+		a = b - a
```
Clever! 🙂 However, you can avoid the need for this trickery by using multiple assignment: `a, b = …`.

vinilapeddireddy replied:
updated

juliaogris replied:
Achievement unlocked: received @anzdaddy `clever` label
(I've been working on this for over a year now to no avail).

Commit:
Addressed review comments

1. Optimised loop 2.As per review comment modified multiple assignment

juliaogris commented on 01_fibonacci/fibi.go:
```diff
@@ -0,0 +1,25 @@
+	a, b := 0, 1
+
+	for i := 0; i <= n; i++ {
+		if a >= 1 {
```
You can get rid of the `if` by changing the loop initialisation.

anzboi replied:
As per @juliaogris comment, try and get rid of the need for the `if` statement

juliaogris commented on 01_fibonacci/fibi_test.go:
```diff
@@ -0,0 +1,24 @@
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
```
Let's test `fibonacci ` function as well (unit tests ;) ) and include some corner cases:
```go
n :=  0
n := 1
n := -1
n := // find a very large number that causes a stackoverflow and comment....

anzboi replied:
As per @juliaogris comment, please add some more tests.

juliaogris reviewed (commented):
This looks really good!
Very close....

juliaogris commented:
Two more comments:
- Please fix up PR description
- Please edit your commits to follow 
   * Capitalize the subject line
   * Use the imperative mood in the subject line
   * Fold fixup commits into previous ones with `git rebase -i`

I'm happy to give a hand if needed.

anz-rfc commented on 01_fibonacci/fibi.go:
```diff
@@ -0,0 +1,25 @@
+
+	for i := 0; i <= n; i++ {
+		if a >= 1 {
+			fmt.Fprintln(out, a)
```
better to decouple fibonacci sequence calculation from IO

https://github.com/anz-bank/go-training/issues/37

vinilapeddireddy replied:
@anz-rfc it's good suggestion.It can be done using arrays to decouple fibonacci sequence calculation from IO, after completing the arrays lesson, will re work on this

anzboi commented on 01_fibonacci/fibi.go:
```diff
@@ -0,0 +1,25 @@
+		if a >= 1 {
+			fmt.Fprintln(out, a)
+		}
+		a, b = b, (a + b)
```
No need to put `a + b` in parens

Commit:
Merge branch 'master' of https://github.com/anz-bank/go-training

Commit:
Merge branch 'master' of https://github.com/anz-bank/go-training

mohankrishna committed:
Implement a function that prints first n Fibonacci numbers

* Update Fibonacci logic to print Negafibonacci numbers for negative n

juliaogris reviewed (commented):
-   Click `Update branch` and locally `git pull origin master` before adding new commits.
-  PR should be issued from feature branch not master, never commit to master
-   Please fix up PR description according to [template](https://github.com/anz-bank/go-training/blob/master/.github/pull_request_template.md)
-   Please edit your commits to follow [recommendations](https://chris.beams.io/posts/git-commit/)
    -   Capitalize the subject line
    -   Use the imperative mood in the subject line
    -   Fold fixup commits into previous ones with `git rebase -i HASH`
I can give a hand with this if you want

madhavansu committed:
Implement Bubble, Insertion sort and a unit test coverage (#41)

* Add bubble sort function and unit tests

* Add insertion sort and unit tests

* Update the folder structure

* Address the review comments

* Update the expected input and tests

* Test the insertion sort

* Remove blank line

* Update the CI

* Modify DeepEqual to Equal to avoid additional package

eobjects committed:
Bubble Sort Implementation (#44)

* Bubble Sort Implementation

* Refactor to fix code review comments and Added test cases

* Added more test cases for negative scenarios

* Refactor to optimise bubble sort and also Moved the code to user subdirectory

* Refactor to fix review comments

* Refactored to fix review comment on initialisation

* Formatted few lines

mohankrishna committed:
Letter frequency (#66)

* Implement a method that returns a mapping of each letter to its frequency

* Implement a method that returns a sorted slice of strings with elements "{key}:{val}" from letter map

Commit:
Remove fibonaci folder from master

sahidakhtar1 committed:
Lab2 bubble sort (#61)

* Add method to perform bubble sort on a list of `int`.

* Add Unit test cases to test bubble sort using table driven testing.

* Create folder structure 02_bubble/USERNAME

* Use direct string literals instead of `strconv.Quote`

* Use string literals directly instead of strconv.Quote

* Change to address review comments.
	- bubbleSort() will return new sorted list without modifying the original list.
	- Test case to test the input value is not modified

* Fix GoLangCI issue.

* Change to update test case to check input values to bubbleSort()

eobjects committed:
Lab3 (#64)

* Initial implementation of Program to map letter frequency to the letter

* Refactor to add test case

* Fixed bot review comment

* Fixed review comments and added couple of test cases

* Added test for non ASCII and spaces

* Refactor to fix review comments

* Refactored tests to use table driven tests

* Refactor to fix review comment on removing rune casting

eobjects committed:
Calculate and print first n Fibonacci numbers and related tests (#60)

* Calculate and print first n Fibonacci numbers and related tests

* Refactor to rename fib function as specified in lab

* Fixed review comments

* Fixed review comment in test

* Fixed test cases

* Refactored to move to user folder and removed additional lines

* Refactor to fix review comment on exit for invalid input

vinilapeddireddy committed:
Lab2 Bubble sort (#56)


* Sort array using bubble sort

*Add unit tests

* Add username to folder

* remove folder

* change typo error

* remove fibonacci from lab2

* Change folder structure

* remove old folder

* Add unit to verify DoesntModifyInput

madhavansu committed:
Lab 4 and 5: Numeronym and Stringers implementation (#84)

* Implement the functionality of numeronyms and stringer

* Implement the functionality to print the IP address in dotted format

* Implement the tests for numeronym

* Remove whitespace

* Add unicode tests validation for numeronym and update to stringer test value

* Change the logical implementation from string append to string pre-allocation

* Rename variable and argument receiver name and update the tests

* Format update

madhavansu committed:
Lab 3 - Letter frequency implementation (#92)

* Implement letter frequency function

* Implement letter frequency logic and tests

* Format code and update the tests

sahidakhtar1 committed:
05 stringer (#94)

* Create type `IPAddr` with `[4]bytes` to create dotted quad.

* Add Unit test case to test the out put of `main()` and `String()` of stringer

* Define type `MyByte` and use to define type `IPAddr`

* Remove `MyByte` and use %d instead of %v.

Commit:
Merge branch 'master' of https://github.com/vinilapeddireddy/go-training

juliaogris commented:
Hi,
There has been  no activity in the last 20 days so I will close this PR.
Feel free to open it again if you have something to add.
Thanks!
Julia
